tests/python/test_household.py

```diff
@@ -0,0 +1,377 @@
+        response = rest_client.get("/us/household/invalid")
+
+        assert response.status_code == 400
+        assert b"Invalid household ID" in response.data
+
+
+class TestCreateHousehold:
+    def test_create_household_success(
+        self, rest_client, mock_database, mock_hash_object
+    ):
+        """Test successfully creating a new household."""
+        # Mock database responses
+        mock_row = MagicMock(spec=LegacyRow)
+        mock_row.__getitem__.side_effect = lambda x: {"id": 1}[x]
+        mock_database.query().fetchone.return_value = mock_row
+
+        response = rest_client.post(
+            "/us/household",
+            json=SAMPLE_HOUSEHOLD_DATA,
+            content_type="application/json",
+        )
+        data = json.loads(response.data)
+
+        assert response.status_code == 201
+        assert data["status"] == "ok"
+        assert data["result"]["household_id"] == 1
+
+    def test_create_household_invalid_payload(self, rest_client):
+        """Test creating a household with invalid payload."""
+        invalid_payload = {
+            "label": "Test",
+            # Missing required 'data' field
+        }
+
+        response = rest_client.post(
+            "/us/household",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Missing required keys" in response.data
+
+    def test_create_household_invalid_label(self, rest_client):
+        """Test creating a household with invalid label type."""
+        invalid_payload = {
+            "data": {},
+            "label": 123,  # Should be string or None
+        }
+
+        response = rest_client.post(
+            "/us/household",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Label must be a string or None" in response.data
+
+
+class TestUpdateHousehold:
+    def test_update_household_success(
+        self, rest_client, mock_database, mock_hash_object
+    ):
+        """Test successfully updating an existing household."""
+        # Mock getting existing household
+        mock_row = MagicMock(spec=LegacyRow)
+        mock_row.__getitem__.side_effect = lambda x: SAMPLE_DB_ROW[x]
+        mock_row.keys.return_value = SAMPLE_DB_ROW.keys()
+        mock_database.query().fetchone.return_value = mock_row
+
+        updated_data = {
+            "data": {"people": {"person1": {"age": 31, "income": 55000}}},
+            "label": "Updated Test Household",
+        }
+
+        response = rest_client.put(
+            "/us/household/1",
+            json=updated_data,
+            content_type="application/json",
+        )
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert data["status"] == "ok"
+        assert data["result"]["household_id"] == "1"
```

Comment:
**question/maybe blocking** Does this actually verify an update happened? I'm expecting a verify of some kind confirming the SQL update.

Reply:
Only in the sense that the API returns a 200 when an update occurs, while it returns a 201 when record creation occurs, but you're right, this test lacks checking beyond that.

Reply:
**suggestion** validate the query unless it's a ton of work to do.